sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -774,11 +851,23 @@ protected Symbol visitSubqueryExpression(SubqueryExpression node, ExpressionAnal
              * But there are no other row-expressions yet. In addition the cast functions and operators don't work with
              * row types (yet).
              *
-             * Since we only support 1 column and only single-row subselects it is okay to use the inner type directly.
+             * However, we support a single column RowType through the SingleColumnTableType.
              */
-            return new SelectSymbol(relation, fields.get(0).valueType());
-        }
+            Field field = fields.get(0);
+            SingleColumnTableType singleColumnTableType = new SingleColumnTableType(field.valueType());
+            Symbol symbolToReturn = new SelectSymbol(relation, singleColumnTableType, node.isArrayExpression());
 
+            // A SubQuery can return more than one row. We don't allow multiple rows,
+            // except in ANY or IN expressions. Thus, we wrap the result into a function
+            // which extracts the first element and checks if there are more than one element.
+            if (node.isArrayExpression() == false) {
```

Comment:
this should not be required, we need to support multiple rows and use the singlecol table type for every subselect that  is a  valueexpression.